silk/unit_tests/mock_device.py

```diff
@@ -0,0 +1,121 @@
+# limitations under the License.
+
+import configparser
+import logging
+import random
+
+from silk.hw.hw_module import HwModule
+from silk.node.fifteen_four_dev_board import FifteenFourDevBoardNode, ThreadDevBoard
+from silk.utils import signal
+
+class MockHwModule(HwModule):
+  """Mock HwModule for unit testing.
+  """
+  def __init__(self, name: str, node_id: int):
+    """Initialize a mock HwModule.
+
+    Args:
+      name (str): name of the node.
+      node_id (int): ID of the node.
+    """
+    super().__init__(name=name,
+                     parser=configparser.ConfigParser(),
+                     node_id=node_id,
+                     port=f"/dev/tty{node_id}")
+    self._model = "Mock"
+    self._dut_serial = "683" + str(random.getrandbits(24))
+    self._layout_center = 100, 100
+    self._layout_radius = 100
+    self._layout_x = None
+    self._layout_y = None
+  
+  def set_otns_vis_position(self, x: int, y: int):
+    """Set mock OTNS visualization position.
+
+    Args:
+      x (int): x coordinate of visualization position.
+      y (int): y coordinate of visualization position.
+    """
+    self._layout_x = x
+    self._layout_y = y
+  
+  def get_otns_vis_position(self):
+    """Get mock OTNS visualization position.
+
+    Raises:
+      ValueError: if OTNS visualization position is not set.
+
+    Returns:
+      Tuple[int, int]: OTNS visualization position coordinates.
+    """
+    if self._layout_x is None or self._layout_y is None:
+      raise ValueError("Node position must have x and y coordinates.")
+
+    return self._layout_x, self._layout_y
+
+  def set_otns_layout_parameter(self, x: int, y: int, radius: int):
+    """Set mock OTNS visualization auto layout parameter.
+
+    Args:
+      x (int): x coordinate of visualization center.
+      y (int): y coordinate of visualization center.
+      radius (int): radius visualization circle.
+    """
+    self._layout_center = x, y
+    self._layout_radius = radius
+
+
+class MockThreadDevBoard(ThreadDevBoard):
+  """Mock ThreadDevBoard for unit testing.
+
+  Attributes:
+    _hwModel (str): device hardware model.
+  """
+  _hwModel = "Mock"
+
+  def __init__(self, name: str, node_id: int):
+    """Initialize a mock ThreadDevBoard.
+
+    Args:
+      name (str): name of the node.
+      node_id (int): ID of the node.
+    """
+    mock_device = MockHwModule(name, node_id)
+    FifteenFourDevBoardNode.__init__(
+        self,
+        virtual=True,
+        device=mock_device,
+        device_path=mock_device.port())
+    if self.logger:
+      self.logger.setLevel(logging.WARN)
+
+
+class MockWpantundProcess(signal.Publisher):
+  """Mock signal publisher that act as wpantund process.
+  """
+  def __init__(self):
+    """Initialize a mock wpantund process.
+    """
+    super().__init__()
+    self._process_id = random.getrandbits(16)
+    self._prefix = f"wpantund[{self._process_id}]: NCP => [OTNS] "
```

Comment:
Please define the Regex in otns_manager module and import from it.

Reply:
`otns_manager` has `RegexType.STATUS = r"wpantund\[(\d+)\]: NCP => .*\[OTNS\] ([\w\d]+=[A-Fa-f0-9,rsdn]+)"`, but I can't simply reuse that regex since this is a string prefix to be added to all mock wpantund logs. I.e. not for matching, but to turn `role=1` into `wpantund[12345]: NCP => [OTNS] role=1`.